### ISSUE TITLE
Java 7 version?

### ISSUE THREAD

**Issue by mpontes** [irenical/shifty#1]
We don't have to bother to support anything other than Java 8 for all of our server libraries, but it might be helpful to have a backport of this for clients that are constrained with an old version of Java, be it legacy platforms that require Java 7 to run or Android clients, which is only source-compatible with 6, IIRC. The downside is that shifty API seems really cumbersome to use without Java 8 lambdas, so this might not be worthwhile in the end.

For the time being, I'm keeping a [Java 6+ compatible fork here](https://github.com/mpontes/shifty-bp).

**Comment by tiagosimao** [comment]
Good point. As of now, lambdas are (only) used in three places, I think the Java6 equivalent API is still simple enough to use. Closer to Hystrix.

**Comment by mpontes** [comment]
Backport synced with latest version. Getting harder. Java 6 is still supported, but tests only compile+run on Java 7+ because of AutoCloseable.

**Comment by dneves** [comment]
If the reason is android, i think it's safe to forget java6 and go with java7. as i've read somewhere, AutoClosable in Java 7 is implemented in Android API level 19.

**Comment by mpontes** [comment]
I noticed that too, but I'm worried that requiring Android 4.4 might be too much for some people.
